### PR TITLE
Use co blocks for connect:auth and connect:setup

### PR DESCRIPTION
**Pull request by gulopine**
This got rearranged a little more than others, because of also moving to `withConnection` instead of `withUserConnections`, because we're only operating on a single connection anyway.

### CODE REVIEW

**Comment by gulopine** [comment]
Connects to heroku/herokuconnect#2612

**Comment by gulopine** [comment]
Connects to heroku/herokuconnect#2612

File: lib/commands/connect/auth_connection.js [hunk 1]
```diff
@@ -27,27 +29,24 @@ module.exports = {
...
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku)
-        .then(function(connections) {
-          if (connections.length > 0) {
-            callbackServer();
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withConnection(context, heroku);
+      callbackServer();
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
```
Comment (iamjem) [inline lib/commands/connect/auth_connection.js]: I believe its safe to use template strings as well (if you want).
``` javascript
let url = `/api/v3/connections/${connection.id}/authorize_url`;
```

Assuming [this feature list](https://nodejs.org/en/docs/es6/#which-features-ship-with-node-js-by-default-no-runtime-flag-required) is a good reference.

File: lib/commands/connect/auth_connection.js [hunk 2]
```diff
@@ -27,27 +29,24 @@ module.exports = {
...
-              });
-            } catch (e) {console.log(e)};
-          } else {
-            console.log("No connection found");
-          }
-        });
-    })
+      console.log("Launching Salesforce for authorization. If your browser doesn't open, please copy the following URL to proceed:\n\n" + redir + "\n");
+      try {
+        open(redir);
```
Comment (iamjem) [inline lib/commands/connect/auth_connection.js]: We can probably get rid of the `open` package as a dependency, and use `heroku-cli-util` instead, which has an [open method of its own](https://github.com/heroku/heroku-cli-util#open-web-browser) (possibly just aliasing this package, I'm not sure) and is also a dependency.

**Comment by iamjem** [comment]
none of my comments are anything critical, :+1:

File: lib/commands/connect/auth_connection.js (hunk 1 above)
Comment (gulopine) [inline lib/commands/connect/auth_connection.js]: That's probably a good idea. For now, I just want to reproduce the existing functionality though, so I'll leave that cleanup for another PR.

File: lib/commands/connect/auth_connection.js (hunk 2 above)
Comment (gulopine) [inline lib/commands/connect/auth_connection.js]: Ah, yes. I had actually prepped another PR to do just that, before remembering that I already had this one in progress. I'll update this to use it, but I'll hold off on removing the depdendency until we've resolved #26, which also uses `open()`.

**Pull request merged**